test/modify.t

```diff
@@ -0,0 +1,158 @@
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+# https://www.opensource.org/licenses/mit-license.php
+#
+###############################################################################
+
+import os
+import sys
+import unittest
+
+from datetime import datetime, timedelta
+
+# Ensure python finds the local simpletap module
+sys.path.append(os.path.dirname(os.path.abspath(__file__)))
+
+from basetest import Timew, TestCase
+
+class TestModify(TestCase):
+    def setUp(self):
+        """Executed before each test in the class"""
+        self.t = Timew()
+
+    def test_modify_open_interval(self):
+        """Attempt to modify an open interval"""
+        now_utc = datetime.now().utcnow()
+        one_hour_before_utc = now_utc - timedelta(hours=1)
+
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(one_hour_before_utc))
+        code, out, err = self.t.runError("modify stop @1 {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc))
+        self.assertIn("Cannot modify open interval", err)
+
+    def test_modify_invalid_subcommand(self):
+        """Modify without (start|stop) subcommand"""
+        now_utc = datetime.now().utcnow()
+        one_hour_before_utc = now_utc - timedelta(hours=1)
+
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(one_hour_before_utc))
+        self.t("stop")
+        code, out, err = self.t.runError("modify @1 {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc))
+        self.assertIn("invalid argument", err)
+
+    def test_modify_no_stop_time(self):
+        """Modify without a time to stop at"""
+        now_utc = datetime.now().utcnow()
+        one_hour_before_utc = now_utc - timedelta(hours=1)
+
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(one_hour_before_utc))
+        self.t("stop")
+        code, out, err = self.t.runError("modify stop @1")
+        self.assertIn("No updated time", err)
+
+    def test_modify_shorten_one_hour(self):
+        """Shorten the interval by one hour."""
+        now_utc = datetime.now().utcnow()
+
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=3)))
+        self.t("stop {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=1)))
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(minutes=49)))
+        self.t("stop")
+
+        code, out, err = self.t("modify stop @2 {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=2)))
+
+        # Verify that the total time is one hour less than it was before.
+        code, out, err = self.t("summary")
```

Comment:
Better use `self.t.export()` than parsing the output of `timew summary`, see e.g. `move.t:117`:
```
j = self.t.export()

self.assertEquals(len(j), 2)
self.assertClosedInterval(j[0],
                          expectedStart="20170301T110000Z",
                          expectedEnd="20170301T133000Z",
                          expectedTags=["foo"])
```

Reply:
Ack